pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -112,7 +112,7 @@ func (c *Controller) Services() ([]*model.Service, error) {
 	// are installed in multiple clusters.
 	smap := make(map[host.Name]*model.Service)
 
-	services := make([]*model.Service, 0)
+	services := make([]*model.Service, 0, len(c.GetRegistries()))
```

Comment:
This seems incorrect

Reply:
Thanks for review. Yup, I verified as there are two loops so length will be dynamic. Done.